File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
"""
leet

ord -> [-1]*26
if not -1 -> len(s)
if -1 idx,

-1 -> len(s)
min()

"""

class Solution:
    def firstUniqChar(self, s: str) -> int:
        array=[-1 for _ in range(26)]
        for i in range(len(s)):
            n=ord(s[i])-97
            if array[n]>-1:
                array[n]=len(s)
            else:
                array[n]=i
        for i in range(26):
            if array[i]==-1:
                array[i]=len(s)
        answer=min(array)
        if answer==len(s):
            return -1
        return answer
        
        

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> charCount;
        unordered_map<char, int> firstIndex;
        int minIndex = numeric_limits<int>::max();
        
        for (int i = 0; i < s.length(); ++i) {
            charCount[s[i]]++;
            if (firstIndex.find(s[i]) == firstIndex.end()) {
                firstIndex[s[i]] = i;
            }
        }
        
        for (const auto& pair : firstIndex) {
            if (charCount[pair.first] == 1) {
                minIndex = min(minIndex, pair.second);
            }
        }
        
        if (minIndex == numeric_limits<int>::max()) {
            return -1;
        }
        
        return minIndex;
    }
};